test/distributed/org/apache/cassandra/distributed/impl/IsolatedJmx.java

```diff
@@ -57,8 +60,8 @@ public class IsolatedJmx
     private JMXServerUtils.JmxRegistry registry;
     private RMIJRMPServerImpl jmxRmiServer;
     private MBeanWrapper.InstanceMBeanWrapper wrapper;
-    private RMIClientSocketFactoryImpl clientSocketFactory;
-    private CollectingRMIServerSocketFactoryImpl serverSocketFactory;
+    private RMIClientSocketFactory clientSocketFactory;
```

Comment:
This is because the other places down the line requires `RMIClientSocketFactory` (or ServerSocketFactory) and `RMICloseableSocketFactory` type won't work there.

Reply:
@maulin-vasavada  I am not sure how complicated it would be to rewokr kthat but if not possible then you should check it with `instanceof` before casting it like that. Now you assume what happens outside of this class. You assume that you will put `RMICloseableSocketFactory` into `socketFactories`.